heudiconv/convert.py

```diff
@@ -523,6 +523,35 @@ def update_uncombined_name(
     return filename
 
 
+def update_multiorient_name(
+    metadata: dict[str, Any],
+    filename: str,
+) -> str:
+    if "acq-" in filename:
+        lgr.warning(
+            "Not embedding multi-orientation information as prefix already uses acq- parameter."
+        )
+        return filename
+    iop = metadata.get("ImageOrientationPatientDICOM")
+    # iop = [round(x) for x in iop]
+    assert isinstance(iop, list)
+    cross_prod = [
+        iop[1] * iop[5] - iop[2] * iop[4],
+        iop[2] * iop[3] - iop[0] * iop[5],
+        iop[0] * iop[4] - iop[1] * iop[3],
+    ]
+    cross_prod = [abs(x) for x in cross_prod]
+    slice_orient = ["sagittal", "coronal", "axial"][cross_prod.index(1)]
+    bids_pairs = filename.split("_")
+    # acq needs to be inserted right after sub- or ses-
```

Comment:
according to
https://github.com/bids-standard/bids-specification/blob/master/src/schema/rules/entities.yaml
it should be e.g. after a possible `_task-`  .  But overall -- reminded me about unfinished 
- https://github.com/nipy/heudiconv/pull/544
and that we have base implementation in
- https://github.com/nipy/heudiconv/blob/383525d7d3bdff7fbe52287aa7ee1d091532e39f/heudiconv/bids.py#L1069

couldn't you use it here?